_studio/shared/asc/CMakeLists.txt

```diff
@@ -28,14 +28,22 @@ set( defs "" )
 set( sources "" )
 set( sources.plus "" )
 
+add_library(asc_avx2 OBJECT ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_avx2_impl.cpp)
+target_compile_options(asc_avx2 PRIVATE -mavx2)
+configure_build_variant(asc_avx2 none)
+
+add_library(asc_sse4 OBJECT ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_sse4_impl.cpp)
+target_compile_options(asc_sse4 PRIVATE -msse4.1)
+configure_build_variant(asc_sse4 none)
+
 list( APPEND sources
     ${CMAKE_CURRENT_SOURCE_DIR}/src/asc.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_avx2_impl.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_c_impl.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/asc_sse4_impl.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/iofunctions.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/motion_estimation_engine.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/src/tree.cpp
+    $<TARGET_OBJECTS:asc_avx2>
```

Comment:
Object libraries should go to the LIBS or LIBS_NOVARIANT variables. Please move.

Reply:
See here: https://cmake.org/cmake/help/v3.5/command/add_library.html#object-libraries

Reply:
@onabiull : do you consider this resolved or you still think we need some change?

Reply:
My bad :)